src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -84,27 +82,27 @@ protected ApiConfiguration Configuration
                     this.GetType(),
                     apiType =>
                     {
-                        IServiceCollection services = new ServiceCollection()
-                            .CutoffPrevious<IQueryExecutor>(DefaultQueryExecutor.Instance);
-                        services = this.ConfigureApi(services);
-                        ApiConfiguratorAttribute.ApplyApiServices(apiType, services);
-
-                        // Copy from pre-build registration.
-                        ApiConfiguration.Configuration(apiType)(services);
-
-                        // Make sure that all convention-based handlers are outermost.
-                        EnableConventions(services, apiType);
-                        if (!services.HasService<ApiBase>())
-                        {
-                            services.AddScoped<ApiHolder>()
-                                .AddScoped(apiType, sp => sp.GetService<ApiHolder>().Api)
-                                .AddScoped(sp => sp.GetService<ApiHolder>().Api)
-                                .AddScoped(sp => sp.GetService<ApiHolder>().Api.Context);
-                        }
-
-                        var configuration = this.CreateApiConfiguration(services);
-                        ApiConfiguratorAttribute.ApplyConfiguration(apiType, configuration);
-                        return configuration;
+                        var builder = new ApiBuilder()
+                            .AddInnerMost(ApiConfiguration.Configure(apiType).Configuration)
```

Comment:
Here's a hypothetical usage of ApiBuilder, we could use it to build a partial configuration, and then merge into other ApiBuilders. Not sure if it's useful enough.